librz/core/ctypes.c

```diff
@@ -1026,22 +1026,78 @@ RZ_IPI bool rz_types_open_file(RzCore *core, const char *path) {
 	return true;
 }
 
-RZ_IPI void rz_types_open_editor(RzCore *core, const char *typename) {
+RZ_IPI bool rz_types_open_editor_all(RzCore *core) {
 	RzTypeDB *typedb = core->analysis->typedb;
-	char *str = rz_core_cmd_strf(core, "tc %s", typename ? typename : "");
-	char *tmp = rz_core_editor(core, "*.h", str);
-	if (tmp) {
-		char *error_msg = NULL;
-		int result = rz_type_parse_string_stateless(typedb->parser, tmp, &error_msg);
-		if (result) {
-			// TODO: remove previous types and save new edited types
-			//rz_type_db_purge(typedb);
-		}
-		if (error_msg) {
-			eprintf("%s\n", error_msg);
-			free(error_msg);
+	RzStrBuf *contents = rz_strbuf_new("");
+	if (!contents) {
+		return false;
+	}
+	// Get the all types
+	RzList *types = rz_type_db_get_base_types(typedb);
+	if (!types || rz_list_empty(types)) {
+		return false;
+	}
+	RzListIter *iter;
+	RzBaseType *t;
+	rz_list_foreach (types, iter, t) {
+		char *str = rz_type_db_base_type_as_string(typedb, t);
+		if (!str) {
+			continue;
 		}
-		free(tmp);
+		rz_strbuf_appendf(contents, "%s\n\n", str);
+	}
+	char *str = rz_strbuf_drain(contents);
+	char *tmp = rz_core_editor(core, NULL, str);
+	if (!tmp) {
+		// We removed all types, purge the database
+		rz_type_db_purge(typedb);
+		free(str);
+		return true;
+	}
+	char *error_msg = NULL;
+	bool success = true;
+	int result = rz_type_parse_string_stateless(typedb->parser, tmp, &error_msg);
+	if (result && error_msg) {
+		RZ_LOG_ERROR("%s\n", error_msg);
+		free(error_msg);
+		success = false;
+	}
+	free(tmp);
+	free(str);
+	return success;
+}
+
+RZ_IPI bool rz_types_open_editor_type(RzCore *core, RZ_NONNULL const char *name) {
+	rz_return_val_if_fail(name, false);
+
+	RzTypeDB *typedb = core->analysis->typedb;
+	RzBaseType *btype = rz_type_db_get_base_type(typedb, name);
+	if (!btype) {
+		RZ_LOG_ERROR("Cannot find type \"%s\"\n", name);
+		return false;
+	}
+	char *str = rz_type_db_base_type_as_string(typedb, btype);
```

Comment:
same here... I'd expect things with multiline and better printed in the vim buffer. With this function we get everything in one line.